lib/cisco_node_utils/fabricpath_topology.rb

```diff
@@ -0,0 +1,150 @@
+# Fabricpath Topology provider class
+#
+# Deepak Cherian, November 2015
+#
+# Copyright (c) 2015 Cisco and/or its affiliates.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require_relative 'node_util'
+require_relative 'fabricpath_global'
+
+module Cisco
+  # node_utils class for fabricpath_topology
+  class FabricpathTopo < NodeUtil
+    attr_reader :topo_id
+
+    def initialize(topo_id, instantiate=true)
+      @topo_id = topo_id.to_s
```

Comment:
Why the to_s if you want it to be an integer? Consider:

```
@topo_id = topo_id.to_i
fail ArgumentError unless @topo_id > 0
```

You also won't need the regex check if you use that.